### DISCUSSION TITLE
How to get a JSON name of a field?

### DISCUSSION THREAD

**Discussion by alekangelov** [go-playground/validator#1204] [Q&A] [answered]
error.Field() does not seem to work properly, just gives me the StructField() unless I do RegisterTagNameFunc. Is there any way to do it natively since in the docs it says Field should do this? Am I missing something?

```go

struct Something {
    Field string `json:"field" validate:"required"`
}

validator.Struct(something)

err.Field() => "Field"

```

I must be missing something right? docs say that FIeld() gives me `JSON name "fname"`

Cheers and thank you

**Comment by deankarn** [comment, marked as answer]
The document does not say it will return the JSON name of the field, it is used as an example only and clearly states it will use the field(structs) field name unless overridden with the tag name function see https://github.com/go-playground/validator/blob/master/errors.go#L118

It is working as intended.

I am assuming you are referring to the Go documentation.

**Reply by alekangelov** [reply]
ok sick thanks, i was under the impression that it does this out of the box, my bad 🖐️
